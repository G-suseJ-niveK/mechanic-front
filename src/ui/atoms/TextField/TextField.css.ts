import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  default: {
    '& .Mui-disabled': {
      '&:hover fieldset': {
        borderColor: 'rgba(0, 0, 0, 0.31) !important'
      },
      color: 'rgba(0, 0, 0, 0.20)'
    },
    '& .MuiInput-underline:after': {
      borderBottomColor: 'green'
    },
    '& .MuiOutlinedInput-root': {
      '& fieldset': {
        borderColor: '#02A460'
      },
      '&.Mui-focused fieldset': {
        borderColor: '#02A460'
      },
      '&:hover fieldset': {
        borderColor: '#02A460'
      }
    },
    '& label': {
      color: 'rgba(0, 0, 0, 0.65)'
    },
    '& label.Mui-focused': {
      color: 'green'
    },
    '&:hover label': {
      // color: '#8bf333'
    },
    background: 'white',
    borderRadius: '5px',
    marginTop: '8px'
  },
  root: {
    '& .MuiInputLabel-outlined': {
      transform: 'translate(14px, 10px) scale(1)'
    },
    '& .MuiInputLabel-outlined.MuiInputLabel-shrink': {
      transform: 'translate(14px, -6px) scale(0.75)'
    },
    '& .MuiOutlinedInput-input': {
      padding: '9px'
    }
  }
}));

export default useStyles;
