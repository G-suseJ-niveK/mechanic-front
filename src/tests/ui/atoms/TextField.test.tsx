import React from 'react';
import { render, screen, fireEvent, cleanup, queryByAttribute } from '@testing-library/react';
import TextField from '~atoms/TextField/TextField';

afterEach(cleanup);

describe('Render TextField', () => {
  const id: string = 'test-id';
  const name: string = 'test-name';
  const label: string = 'test-label';
  const value: string = 'test-value';
  const testId = 'TextField';

  test('should render', () => {
    const { container } = render(<TextField id={id} name={name} value={value} label={label} />);
    expect(container).toMatchSnapshot();
  });

  test('should have a id, name, label, value and testId', () => {
    const getById = queryByAttribute.bind(null, 'id');
    const getByName = queryByAttribute.bind(null, 'name');

    const { container } = render(<TextField id={id} name={name} value={value} label={label} />);
    expect(getById(container, id)).toBeInstanceOf(Node);
    expect(getByName(container, name)).toBeInstanceOf(Node);
    expect(screen.getByRole('textbox', { name: label })).toBeInstanceOf(Node);
    expect(screen.getByRole('textbox', { name: label })).toHaveValue(value);
    expect(screen.getByTestId(testId)).not.toBeNull();
  });

  test('should change value', () => {
    let changeValue: string = 'prev-text';
    const mockOnChange = jest.fn((e: any) => {
      changeValue = e.target.value;
    });

    const { rerender } = render(
      <TextField id={id} name={name} value={changeValue} label={label} onChange={mockOnChange} />
    );

    expect(screen.getByRole('textbox', { name: label })).toHaveValue(changeValue);
    fireEvent.change(screen.getByRole('textbox', { name: label }), { target: { value: 'newValue' } });

    rerender(<TextField id={id} name={name} value={changeValue} label={label} />);
    expect(screen.getByRole('textbox', { name: label })).toHaveValue('newValue');
  });

  test('should display error message', () => {
    const errors: any = {
      'test-name': 'ERRORS'
    };

    const touched: any = {
      'test-name': true
    };

    render(<TextField id={id} name={name} value={value} label={label} errors={errors} touched={touched} />);
    //container.className.includes('Miu-error')

    expect(screen.getByText('ERRORS')).toHaveClass('Mui-error');
    expect(screen.getByText('ERRORS')).toBeInstanceOf(Node);
  });
});
