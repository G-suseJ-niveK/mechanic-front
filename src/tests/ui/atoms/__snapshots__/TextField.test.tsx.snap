// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TextField should render 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-2 makeStyles-default-1 MuiFormControl-marginNormal MuiFormControl-fullWidth"
    data-testid="TextField"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
      data-shrink="true"
      for="test-id"
      id="test-id-label"
    >
      test-label
    </label>
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input"
        id="test-id"
        name="test-name"
        type="text"
        value="test-value"
      />
    </div>
    
  </div>
</div>
`;
